src/Amazon.Kinesis.Firehose/Actions/PutRecordBatchRequest.cs

```diff
@@ -14,4 +16,41 @@ public PutRecordBatchRequest(string deliveryStreamName, params Record[] records)
 
         public Record[] Records { get; }
     }
+
+    public class PutRecordBatchRequestConverterFactory : JsonConverterFactory
+    {
+        public override bool CanConvert(Type typeToConvert)
+        {
+            return typeToConvert == typeof(PutRecordBatchRequest);
+        }
+
+        public override JsonConverter CreateConverter(Type typeToConvert, JsonSerializerOptions options)
+        {
+            return new PutRecordBatchRequestConverter();
+        }
+    }
+
+    public class PutRecordBatchRequestConverter : JsonConverter<PutRecordBatchRequest>
+    {
+        public override PutRecordBatchRequest Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            throw new NotImplementedException();
+        }
+
+        public override void Write(Utf8JsonWriter writer, PutRecordBatchRequest value, JsonSerializerOptions options)
+        {
+            writer.WriteStartObject();
+            writer.WriteString("DeliveryStreamName", value.DeliveryStreamName);
+            writer.WriteStartArray("Records");
+            foreach (var record in value.Records)
+            {
+                writer.WriteStartObject();
+                writer.WriteBase64String("Data", record.Data);
+                writer.WriteEndObject();
+            }
+            writer.WriteEndArray();
+            writer.WriteEndObject();
+        }
+    }

```

Comment:
These Converter or ConverterFactory options also did not work. We'd get some other errors about some null ref in some other System.Text.Json scaffolding code for getting the converter going.